doc/detection/Footprint.rst

```diff
@@ -0,0 +1,130 @@
+possibly setting the results back to a ``Footprint``, have convenience methods
+defined in the ``Footprint`` class and are summarised below. Unlike the
+``SpanSets`` class a ``Footprint`` is mutable, such that calling the ``dilate``
+or ``erode`` happen *in place* as shown below.
+
+.. code-block:: python
+
+    from lsst.afw.geom import SpanSet
+    from lsst.afw.detection import Footprint
+
+    # Create a SpanSet and use it to construct a Footprint
+    ss = SpanSet.fromShape(5, offset=(10, 10))
+    foot = Footprint(ss)
+
+    # Grab the SpanSet back from the Footprint and show it is the same
+    spanSetMessage = "SpanSets are the same {}"
+    ssFromFoot = foot.spans
+    print(spanSetMessage.format(ss is ssFromFoot)) # Should print True
+
+    # Modify the Footprint in place (automatically update the internal reference
+    # to the SpanSet), show that they are now different
+    foot.erode(2)
+    newSsFromFoot = foot.spans
+    print(spanSetMessage.format(ss is newSsFromFoot)) # should print false
+
+In cases where the location information is all that is needed, it is strongly
+suggested to use a ``SpanSet`` directly and avoid the overhead of carrying
+around an empty ``PeakCatalog``. The constructors for a ``Footprint`` have been
+structured to remind users of this by first requiring a SpanSet to be created.
+
+Peaks
+=====
+
+Information on the location and intensity of the detected peaks within a
+``Footprint`` are kept in a ``PeakCatalog`` which can be retrieved with the
+``getPeaks`` method (or the .peaks property within python). Like with a
+``Footprint's`` ``SpanSet`` it is possible to set a ``Footprint's``
+``PeakCatalog``. Unlike with the ``SpanSet`` member it is only possible to set a
+new ``PeakCatalog`` if the existing catalog is empty. It is also possible to
+change the schema that defines the ``PeakCatalog`` but the existing catalog must
+be empty for this operation as well. Another asymmetry between the ``SpanSet``
+and ``PeakCatalog`` members is in the behavior of the python property accessor.
+In python the .spans property is both read and writable, while the .peaks
+property is read only. This behavior is intended to make a programmer cognisant
+that the ``PeakCatalog`` can only be set if it is currently empty. The
+``PeakCatalog`` can be populated by calling methods on the catalog itself, or
+with the convince method ``addPeak`` supplied with the ``Footprint`` class. An
+additional convince method ``sortPeaks`` is also provided to increase the ease
+of sorting the catalog. The following is an example of using a ``Footprint``'s
+``PeakCatalog``.
+
+.. code-block:: python
+
+    from lsst.afw.geom import SpanSet
+    from lsst.afw.detection import Footprint
+
+    # Create a Footprint from a SpanSet
+    ss = SpanSet.fromShape(5, offset=(10, 10))
+    foot = Footprint(ss)
+
+    # Add a few peaks to the PeakCatalog
+    foot.addPeak(7, 7, 95)
+    foot.addPeak(8, 8, 103)
+    foot.addPeak(9, 9, 100)
```

Comment:
More magic numbers (perhaps add a comment saying what they mean (e.g. "Add a few peaks `(x, y, intensity)` to the ...".